Dockerfile

```diff
@@ -19,156 +20,330 @@
 # SPDX-License-Identifier: Apache-2.0
 # License-Filename: LICENSE
 
-# Set this to the version ORT should report.
-ARG ORT_VERSION="DOCKER-SNAPSHOT"
+#------------------------------------------------------------------------
+# Use OpenJDK Eclipe Temurin Ubuntu LTS
+FROM eclipse-temurin:11-jdk-focal AS build
+
+ENV LANG=C.UTF-8
+
+# Prepare build environment to use ort scripts from here
+COPY scripts /etc/scripts
 
 # Set this to a directory containing CRT-files for custom certificates that ORT and all build tools should know about.
 ARG CRT_FILES=""
+COPY "$CRT_FILES" /tmp/certificates/
+RUN /etc/scripts/import_proxy_certs.sh \
+    && if [ -n "$CRT_FILES" ]; then \
+    /etc/scripts/import_certificates.sh /tmp/certificates/; \
+    fi
+
+#------------------------------------------------------------------------
+# Ubuntu build toolchain
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    build-essential \
+    clang-9 \
+    clang++-9 \
+    dirmngr \
+    dpkg-dev \
+    git \
+    gnupg \
+    libbluetooth-dev \
+    libbz2-dev \
+    libc6-dev \
+    libexpat1-dev \
+    libffi-dev \
+    libgmp-dev \
+    libgdbm-dev \
+    liblzma-dev \
+    libmpdec-dev \
+    libncursesw5-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    make \
+    netbase \
+    tk-dev \
+    tzdata \
+    uuid-dev \
+    unzip \
+    xz-utils \
+    zlib1g-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+#------------------------------------------------------------------------
+# PYTHON - Build Python as a separate component with pyenv
+FROM build as pythonbuild
+
+ARG CONAN_VERSION=1.44.0
+ARG PYTHON_VERSION=3.9.9
+ARG PYTHON_VIRTUALENV_VERSION=20.13.0
+ARG PIPTOOL_VERSION=21.3.1
```

Comment:
As a general idea for improvement: It would be nice of the installation of a tool could be skipped completely by setting the corresponding build argument to an empty string or so.

Reply:
Unfortunately IF/ELSE is something that docker can't handle.
The "solution" would be a external tool that dynamically create the Dockerfile, but again, will be too much.